src/services/access/useGenerateUserToken.ts

```diff
@@ -0,0 +1,65 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query'
+import { z } from 'zod'
+
+import Api from 'shared/api'
+import { NetworkErrorObject, rejectNetworkError } from 'shared/api/helpers'
+
+import { USER_TOKEN_TYPE } from './constants'
+
+const UseGenerateTokenResponseSchema = z.object({
+  createUserToken: z
+    .object({
+      error: z
+        .union([
```

Comment:
One small nit, can we move this to a `.discriminatedUnion()`, you can read a bit more about them [here](https://zod.dev/?id=discriminated-unions), but they're a super awesome way to union on a single field and makes checking for said field super easy.